tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -939,3 +963,152 @@ def _get_num_gpus_on_instance(instance_type_info):
     }
     """
     return sum([gpu_type.get("Count") for gpu_type in instance_type_info.get("GpuInfo").get("Gpus")])
+
+
+@retry(wait_fixed=seconds(20), stop_max_delay=minutes(5))
+def _wait_for_computefleet_changed(cluster, desired_status):
+    assert_that(cluster.status()).contains(f"ComputeFleetStatus: {desired_status}")
+
+
+@retry(wait_fixed=seconds(20), stop_max_delay=minutes(5))
+def _wait_for_partition_state_changed(scheduler_commands, partition, desired_state):
+    assert_that(scheduler_commands.get_partition_state(partition=partition)).is_equal_to(desired_state)
+
+
+def _update_and_start_cluster(cluster):
+    cluster.stop()
+    _wait_for_computefleet_changed(cluster, "STOPPED")
+    cluster.update()
+    cluster.start()
+    _wait_for_computefleet_changed(cluster, "RUNNING")
+
+
+def _inject_bootstrap_failures(cluster, bucket_name, pcluster_config_reader):
+    """Update cluster to include pre-install script, which introduce bootstrap error."""
+    updated_config_file = pcluster_config_reader(config_file="pcluster.config.update.ini", bucket=bucket_name)
+    cluster.config_file = str(updated_config_file)
+    _update_and_start_cluster(cluster)
+
+
+def _set_protected_failure_count(remote_command_executor, test_datadir, protected_failure_count):
+    """Disable protected mode by setting protected_failure_count to -1."""
+    remote_command_executor.run_remote_command(
+        f"echo 'protected_failure_count = {protected_failure_count}' | sudo tee -a "
+        "/etc/parallelcluster/slurm_plugin/parallelcluster_clustermgtd.conf"
+    )
+    remote_command_executor.run_remote_script(str(test_datadir / "slurm_restart_clustermgtd.sh"), run_as_root=True)
+
+
+def _enable_protected_mode(remote_command_executor, test_datadir):
+    """Enable protected mode by removing lines related to protected mode in the config, so it will be set to default."""
+    remote_command_executor.run_remote_command(
+        "sudo sed -i '/'protected_failure_count'/d' /etc/parallelcluster/slurm_plugin/parallelcluster_clustermgtd.conf"
+    )
+    remote_command_executor.run_remote_script(str(test_datadir / "slurm_restart_clustermgtd.sh"), run_as_root=True)
+
+
+def _test_disable_protected_mode(remote_command_executor, test_datadir):
+    """Test Bootstrap failures have no affect on cluster when protected mode is disabled."""
+    # Disable protected mode by setting protected failure count to -1
+    _set_protected_failure_count(remote_command_executor, test_datadir, -1)
+    # wait till the node failed
+    retry(wait_fixed=seconds(20), stop_max_delay=minutes(15))(assert_errors_in_logs)(
```

Comment:
Do we need to wait 15 mins? If it's just a pre-install failure that causes self-termination it should happen quickly, maybe we can try 5 or 7 mins instead?
Same comment applies to everywhere that we are using the 15 mins timeout

Reply:
Set them to 7 minutes